ui/src/app/views/workspace-edit.tsx

```diff
@@ -360,6 +360,148 @@ export const LabeledCheckBox = (props) => {
 
 export enum WorkspaceEditMode { Create = 1, Edit = 2, Duplicate = 3 }
 
+export interface SearchInputState {
+  matches: Array<string>;
+  hover: Array<boolean>;
+  state: number;
+}
+
+function getDiseaseNames(keyword) {
+  const baseurl = serverConfigStore.getValue().firecloudURL;
+  const url = baseurl + '/duos/autocomplete/' + keyword;
+  return fetch(encodeURI(url)).then((response) => {
+    return response.json();
+  }).then((matches) => {
+    const labeledMatches = fp.filter((elt) => elt.hasOwnProperty('label'))(matches);
+    const diseases = fp.map((elt) => elt['label'])(labeledMatches);
+    return diseases;
+  });
+}
+
+class SearchInput extends React.Component<any, SearchInputState> {
```

Comment:
Can we declare the `Props` explicitly? So we know what inputs this component takes

Reply:
And how about moving this into its own file? We have been trying to stick to 1 component per file where possible. Less stuff to digest per file and better enforces separation between components.

Reply:
Done